sched/scheduler/task_runner.go

```diff
@@ -10,23 +11,23 @@ import (
 // Run the task on the specified worker, and update the SagaLog appropriately.  Returns an error if one
 // occurs while running a task or writing to the SagaLog.  This method blocks until all saga messages
 // are logged and the task completes
-func runTaskAndLog(saga *saga.Saga, worker worker.Worker, taskId string, task sched.TaskDefinition) error {
+func runTaskAndLog(saga *saga.Saga, worker worker.Worker, taskId string, task sched.TaskDefinition) (runner.ProcessStatus, error) {
```

Comment:
suggest just passing the stat field in and logging stats in here. 

The intention is that this function handles everything associated with running a task.  Sending it to the worker, updating saga log, and IMO logging information about tasks.  

There's more granularity here about what went wrong we probably want to have a counter to Log when updating the Saga Fails to differentiate between when a Worker Fails etc... This is the info we'll need to debug the service in production